src/Core/src/Handlers/Button/ButtonHandler.iOS.cs

```diff
@@ -166,20 +164,48 @@ static void SetControlPropertiesFromProxy(UIButton platformView)
 			}
 		}
 
-		void OnButtonTouchUpInside(object? sender, EventArgs e)
+		class ButtonEventProxy
 		{
-			VirtualView?.Released();
-			VirtualView?.Clicked();
-		}
+			WeakReference<IButton>? _virtualView;
 
-		void OnButtonTouchUpOutside(object? sender, EventArgs e)
-		{
-			VirtualView?.Released();
-		}
+			IButton? VirtualView => _virtualView is not null && _virtualView.TryGetTarget(out var v) ? v : null;
```

Comment:
@PureWeen  should this rather be at the base level for all handlers? Why are handlers holding strong references to virtual views when the virtual view decides what hander it is? Or is it the other way around?

At least for iOS, we should do this at the base level so that we can avoid this since all controls with events will do this.

Reply:
I think we should perhaps do both? From other cases where we've passed something into these special classes, we usually pass the handler in and then make it a weak reference to the handler. 

For example [this](https://github.com/dotnet/maui/blob/b324e2c473b19f1fa30d07e325bfc0241f663e76/src/Core/src/Handlers/DatePicker/DatePickerHandler.iOS.cs#L131). 

This special class is going to have to hold a weak reference to something. So, I don't think generalizing this inside the base of the handler will really change much of the code here.